### ISSUE TITLE
Obsidian - Malicious pool deployer can set a malicious interest rate contract to lock funds of vault depositors

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-06-new-scope-judging#233]
Obsidian

High

# Malicious pool deployer can set a malicious interest rate contract to lock funds of vault depositors

### Summary

Once vault depositors have deposited funds into a pool, a malicious pool creator can upgrade the `interestRateStrategy` contract (using `PoolConfigurator.setReserveInterestRateStrategyAddress()` to make all calls to it revert.

As a result any function of the protocol that calls `updateInterestRates()` will revert because `updateInterestRates()` makes the following [call](https://github.com/sherlock-audit/2024-06-new-scope/blob/c8300e73f4d751796daad3dadbae4d11072b3d79/zerolend-one/contracts/core/pool/logic/ReserveLogic.sol#L160-L171):
```solidity
(vars.nextLiquidityRate, vars.nextBorrowRate) = IReserveInterestRateStrategy(_reserve.interestRateStrategyAddress)
      .calculateInterestRates(
      /* PARAMS */
    );
```

The main impact is that now withdrawals will revert because `executeWithdraw()` calls `updateInterestRates()` which will always revert, so the funds that vault users deposited into this pool are lost forever.

### Root Cause

Allowing the pool deployer to specify any `interestRateStrategyAddress`

### Attack Path

1. Vault users deposit into the pool
2. the deployer sets their `interestRateStrategy` contract to make all calls to it revert
3. All calls to withdraw funds from the pool will revert, the vault depositors have lost their funds

### Impact

All the funds deposited to the pool from the vault will be lost

### Mitigation

Use protocol whitelisted interest rate calculation contracts

**Comment by iamnmt** [comment]
Escalate

Per the contest's `README`

> There are two set of actors. Actors who manage pools and actors who mange vaults. If an action done by one party causes the other party to suffer losses we'd want to consider that.

This statement makes this issue valid.

**Comment by sherlock-admin3** [comment]
> Escalate
> 
> Per the contest's `README`
> 
> > There are two set of actors. Actors who manage pools and actors who mange vaults. If an action done by one party causes the other party to suffer losses we'd want to consider that.
> 
> This statement makes this issue valid.

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by zarkk01** [comment]
IMO, the issue is **invalid** due to this statement of the sponsor.

> Essentially we expect all permissioned actors to behave rationally.

It is, absolutely, irrational for a deployer to set a malicious interest rate contract since he has **nothing** to earn out of this behaviour.

**Comment by 0xSpearmint** [comment]
The permissioned actors the protocol refers to are the protocol controlled `PoolConfigurator` and `owner` roles. They can be expected to act rationally. 

This issue involves a malicious pool deployer (which can be anyone).

Deploying pools is permission-less, which is why the protocol was interested in such issues as they clearly stated in the README:
>There are two set of actors. Actors who manage pools and actors who mange vaults. If an action done by one party causes the other party to suffer losses we'd want to consider that.

**Comment by cvetanovv** [comment]
I agree with the escalation of this issue to be High severity. For more information on what I think about the rule, you can see this comment: https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/234#issuecomment-2413297520

Planning to accept the escalation and make this issue High severity.

**Comment by WangSecurity** [comment]
Result:
High 
Has duplicates

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [iamnmt](https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/233/#issuecomment-2391294265): accepted

**Comment by DemoreXTess** [comment]
@cvetanovv Can we reconsider this issue per this comment : https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/234#issuecomment-2427578837_

The report wrongly states that the funds are locked forever. ZeroLend has permission to make changes on the pools. Users can get back their funds after adjustment by ZeroLend

**Comment by 0xSpearmint** [comment]
The referenced [comment ](https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/234#issuecomment-2427578837)is not accurate.

Even if the protocol sets a new IRM through [this only configurator function](https://github.com/zerolend/zerolend-one/blob/6b681f2a16be20cb2d43e544c164f913a8db1cb8/contracts/core/pool/Pool.sol#L150-L158), the pool admin can instantly change it back to the malicious IRM using [this only pool admin function](https://github.com/zerolend/zerolend-one/blob/6b681f2a16be20cb2d43e544c164f913a8db1cb8/contracts/core/pool/manager/PoolConfigurator.sol#L105-L111).

**Comment by cvetanovv** [comment]
I agree with @0xSpearmint comment. Even if ZeroLend makes any changes, the malicious pool admin can immediately roll back the previous configuration.

**Comment by DemoreXTess** [comment]
@cvetanovv  
Is there a problem in this one, it says escalation resolved with has duplicates but the label is not added.

**Comment by cvetanovv** [comment]
@DemoreXTess That's not a problem. The labels of duplicate issues will be added after all escalations are resolved.

**Comment by Joshuajee** [comment]
@cvetanovv and @WangSecurity ,

Sorry, this might be coming late, but I don't believe that this issue meets the criteria of becoming a High because;

1. It relies on the pool deployer becoming Malicious, what are the chances?
2. It also relies on the vault managers not doing their due diligence on checking if the pool's strategy is indeed safe.
3. Vault Managers are risk managers and should do their thorough checks on every pool, including oracles and strategy contracts before adding such. 

The whole reason for validating this is that the pool deployer can hurt the vault manager, but the vault manager should be rational enough to check the rate strategy contract, before trusting a pool.

**Comment by cvetanovv** [comment]
> @cvetanovv and @WangSecurity ,
> 
> Sorry, this might be coming late, but I don't believe that this issue meets the criteria of becoming a High because;
> 
> 1. It relies on the pool deployer becoming Malicious, what are the chances?
> 2. It also relies on the vault managers not doing their due diligence on checking if the pool's strategy is indeed safe.
> 3. Vault Managers are risk managers and should do their thorough checks on every pool, including oracles and strategy contracts before adding such.
> 
> The whole reason for validating this is that the pool deployer can hurt the vault manager, but the vault manager should be rational enough to check the rate strategy contract, before trusting a pool.

1. The pool deployer does not need to become malicious. He is malicious by default if it can hurt the Vault.
2. I agree here, and every Vault Manager should do their own checking, but it doesn't matter because, at any time, he can change the interest rate because he is malicious.
3. Same as point 2. They are malicious by default, and even if the vault managers do the best check, the pool manager can increase the interest rate at any time.

**Comment by Joshuajee** [comment]
@cvetanovv,

I agree you are totally right here.

I think that my earlier complaint only affects https://github.com/sherlock-audit/2024-06-new-scope-judging/issues/234

Because oracles can not be changed after deployment
